src/docs/MLOS/MLOSUserGuide.adoc

```diff
@@ -166,6 +166,426 @@ E. Click the *Tasks* tab, select an *Preview_Results* task and click on the *Pre
+ selected as the default language.
+The supported programming languages are:
+
+* Linux_Bash
+* Windows_Cmd
+* DockerCompose
+* Scalaw
+* Groovy
+* Javascript
+* Jython
+* Python
+* Ruby
+* Perl
+* PowerShell
+* R
+
+Here is an example that shows a task implementation written in `Linux_Bash`:
+
+```bash
+#%task(name=myTask, language=Linux_Bash)
+echo 'Hello, World!'
+```
+
+===== Dependencies
+
+One of the most important notions in workflows is the dependencies between tasks. To specify this information, use the
+`dep` parameter. Its value should be a list of all tasks on which the new task depends. Example:
+
+```python
+#%task(name=myTask,dep=[parentTask1,parentTask2])
+print('Hello world')
+```
+
+===== Generic information
+
+To specify the values of some advanced ProActive variables called
+https://doc.activeeon.com/latest/user/ProActiveUserGuide.html#_generic_information[Generic Information], you should the
+provide the `generic_info` parameter. Its value should be a list of tuples `(key,value)` that corresponds the names and
+adequate values of the Generic Information. Example:
+
+```python
+#%task(name=myTask, generic_info=[(var1,value1),(var2,value2)])
+print('Hello world')
+```
+
+===== Export/import variables
+
+The `export` and `import` parameters make variables propagation possible between the different tasks of a workflow.
+If `myTask1` variables `var1` and `var2` are needed in `myTask2`, both pragmas have to specify this information as follow:
+
+* `myTask1` should include an `export` parameter with a list of these variable names,
+* `myTask2` should include an `import` parameter with a list including the same names.
+
+Example:
+
+`myTask1` implementation bloc would be:
+```python
+#%task(name=myTask1, export=[var1,var2])
+var1 = "Hello"
+var2 = "ActiveEon!"
+```
+
+and `myTask2` implementation bloc would be:
+```python
+#%task(name=myTask2, dep=[myTask1], import[var1,var2])
+print(var1 + " from " + var2)
+```
+
+===== Implementation file
+
+It is possible to use an external implementation file as a task implementation. To do so, the option `path` should be used.
+Example:
+
+```python
+#%task(name=myTask,path=PATH_TO/IMPLEMENTATION_FILE.py)
+```
+
+==== Importing libraries
+
+The main difference between ProActive kernel and a native language one resides in the memory access during the
+execution of the different blocs.  While in a common native language kernel the whole script code (all the notebook
+blocs) is executed locally in the same shared memory zone, the ProActive kernel each created task will be executed as
+an independent process.
+To facilitate the transition from native language kernels to the ProActive one we included the pragma `#%import()`.
+This pragma gives the ability to the user to add, just once in the notebook, libraries that are common to all created
+tasks implemented in a same native script language.
```

Comment:
The main difference between the ProActive and 'native language' kernels resides in the way the memory is accessed during blocs execution. While in a common native language kernel, the whole script code (all the notebook blocs) is locally executed in the same shared memory space, the ProActive kernel will execute each created task in an independent process. In order to facilitate the transition from native language to ProActive kernels, we included the pragma #%import(). This pragma gives the user the ability to add libraries that are common to all created tasks implemented in the same native script language.

Not sure about the way I refactored last sentence

Reply:
it's "blocks"... you french speakers ;)

I agree that this paragraph requires some refactoring, I would just remove the comma in  "kernel, the", which actually breaks the flow of the sentence when it shouldn't be the case.
I propose another wording of the sentence above, using "; whereas" for contrast, which also splits the sentence in two more readable parts. Up to @mklkun the final result.

Reply:
And now?

```The main difference between the ProActive and 'native language' kernels resides in the way the memory is accessed during blocks execution. In a common native language kernel the whole script code (all the notebook blocks) is locally executed in the same shared memory space; whereas the ProActive kernel will execute each created task in an independent process. In order to facilitate the transition from native language to ProActive kernels, we included the pragma #%import(). 
This pragma gives the user the ability to add libraries that are common to all created tasks, and thus relative distributed processes,  that are implemented in the same native script language.```

Reply:
all instead of thus